File: day6.py
f = open('./input6.txt').read()

for i in range(len(f)):
    marker_list = f[i:i+4]
    marker_set = set(marker_list)

    message_list = f[i:i+14]
    message_set = set(message_list)

    if len(marker_list) == len(marker_set): print('Marker end:', i+4, marker_list);
    if len(message_list) == len(message_set): print('Message marker end:', i+14, message_list); break

File: day6.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func Day6() {

    file, err := os.Open("./input6.txt")
    if err != nil {
        panic(err)
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    scanner.Scan()
    signalBuffer := scanner.Text()

    for index := range signalBuffer {
        markerTest := NewSet()
        markerTest.bulkAdd(signalBuffer[index:index+14])

        count := 0
        for _, val := range markerTest.contains {
            if val == true { count++ }
        }

        if count == 14 {
            fmt.Println(index+14)
            return
        }
    }
}
